src/Interpreters/ProcessList.cpp

```diff
@@ -86,15 +87,14 @@ ProcessList::EntryPtr ProcessList::insert(const String & query_, const IAST * as
                 throw Exception("Too many simultaneous queries. Maximum: " + toString(max_size), ErrorCodes::TOO_MANY_SIMULTANEOUS_QUERIES);
         }
 
-        String query_kind{ast->getQueryKindString()};
         if (!is_unlimited_query)
         {
-            auto amount = getQueryKindAmount(query_kind);
-            if (max_insert_queries_amount && query_kind == "Insert" && amount >= max_insert_queries_amount)
+            QueryAmount amount = getQueryKindAmount(ast->getQueryKind());
```

Comment:
Better to move ast->getQueryKind() in some variable, because otherwise compiler do not optimize virtual call here.